skyportal/handlers/api/source.py

```diff
@@ -274,13 +290,20 @@ def get(self, obj_id=None):
             )
         if has_tns_name in ['true', True]:
             q = q.filter(Obj.altdata['tns']['name'].isnot(None))
-
+        if group_id:
+            if not group_id.isdigit() or int(group_id) not in [
+                g.id for g in self.current_user.accessible_groups
+            ]:
+                return self.error(
+                    "Requested group '" + group_id + "' is inaccessible to user."
+                )
+            q = q.filter(Source.group_id == int(group_id))
```

Comment:
This should be a list, so the filter expression will change slightly here

Reply:
and done